### PR TITLE
testcase(svc_chaos): util for restarting kubelet/docker services, and, job for app pod to use this util

### PR DESCRIPTION
**Pull request by vishnuitta**
This PR contains util to restart node level services using a daemonset, and, have an env to set the chaos duration.
Also, this PR have litmus job to restart docker service in the node where application pod resides.

Signed-off-by: Vishnu Itta <vitta@mayadata.io>

### CODE REVIEW

File: apps/percona/chaos/openebs_app_svc_chaos/run_litmus_test.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: openebs-app-svc-chaos-
+  namespace: litmus 
+spec:
+  template:
+    metadata:
+      labels:
+        name: openebs-app-svc-chaos-
```
Comment (ksatchit) [inline apps/percona/chaos/openebs_app_svc_chaos/run_litmus_test.yml]:
```suggestion
        name: openebs-app-svc-chaos
```

File: chaoslib/service_chaos/svc_chaos.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      register: svc_chaos_pod
+
+    - name: do svc restart
+      shell: >
+        timeout 10 kubectl exec -it {{ svc_chaos_pod.stdout}} -n {{ namespace }} 
+        -- bash -c "systemctl stop kube-docker-monitor && systemctl stop kubelet-monitor && systemctl stop {{ svc_chaos }} && sleep {{ c_duration }} && systemctl start {{ svc_chaos }} && systemctl start kube-docker-monitor && systemctl start kubelet-monitor"
+      args:
+        executable: /bin/bash
+      ignore_errors: true 
+      register: result
```
[the remark this answers is not in the document]
Comment (vishnuitta) [reply chaoslib/service_chaos/svc_chaos.yml]: ok.. will add checks for status using systemctl command

File: chaoslib/service_chaos/svc_chaos.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    - name: wait for svc chaos duration time
+      shell: >
+        sleep {{ c_duration }} 
+      args:
+        executable: /bin/bash
```
[the remark this answers is not in the document]
Comment (vishnuitta) [reply chaoslib/service_chaos/svc_chaos.yml]: there is no sleep involved in the chaos step. Its a background process

File: chaoslib/service_chaos/svc_chaos.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        -n {{ namespace }} | grep {{ app_node }} 
+        | awk '{print $1}'
+      args:
+        executable: /bin/bash
+      register: svc_chaos_pod
+
+    - name: do svc restart
+      shell: >
+        timeout 10 kubectl exec -it {{ svc_chaos_pod.stdout}} -n {{ namespace }} 
+        -- bash -c "systemctl stop kube-docker-monitor && systemctl stop kubelet-monitor && systemctl stop {{ svc_chaos }} && sleep {{ c_duration }} && systemctl start {{ svc_chaos }} && systemctl start kube-docker-monitor && systemctl start kubelet-monitor"
```
Comment (mittachaitu) [inline chaoslib/service_chaos/svc_chaos.yml]: `systemctl stop kube-docker-monitor && systemctl stop kubelet-monitor` are platform dependent. In AWS cluster(ubuntu image) these services are not available to perform `svc_chaos`. 
```
ubuntu@ip-10-0-1-79:~$  sudo systemctl stop kube-docker-monitor
Failed to stop kube-docker-monitor.service: Unit kube-docker-monitor.service not loaded.
ubuntu@ip-10-0-1-79:~$ sudo systemctl stop kubelet-monitor
Failed to stop kubelet-monitor.service: Unit kubelet-monitor.service not loaded.
```
Is it better to pass other services(kube-docker-monitor,kubelet-monitor) from utils like svc_chaos?

Comment (ksatchit) [reply chaoslib/service_chaos/svc_chaos.yml]: good point @mittachaitu ! To find/arrive on the allied services to stop/start with docker - the test will need to be run on multiple platforms. We should be OK to take this for now IMO & conditionally pass the additional services to the svc_chaos utils when we have the details.

**Comment by ksatchit** [comment]
Based on the above discussion, taking this PR. This litmusbook (chaoslib util for svc chaos) can be enhanced to take the routine for `docker stop` based on platform & also for adding checks to ensure chaos *has* occured.

**Pull request merged**